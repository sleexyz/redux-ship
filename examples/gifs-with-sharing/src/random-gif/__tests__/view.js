// @flow
import React from 'react';
import * as Test from '../../test';
import * as RandomGifModel from '../model';
import RandomGif from '../view';

const defaultProps = {
  dispatch: Test.dispatch,
  state: RandomGifModel.initialState,
  tag: 'cat',
};

Test.snapshotComponent(RandomGif, {
  'default': defaultProps,
  loaded: {
    ...defaultProps,
    state: {
      ...defaultProps.state,
      gifUrl: 'https://media2.giphy.com/media/m7ychnf9zOVm8/giphy.gif',
      isLoading: false,
    },
  },
  loading: {
    ...defaultProps,
    state: {
      ...defaultProps.state,
      isLoading: true,
    },
  },
});
