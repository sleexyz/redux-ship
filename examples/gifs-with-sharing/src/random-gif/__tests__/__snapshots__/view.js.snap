exports[`test default 1`] = `
<div
  className="RandomGif">
  <img
    alt="no gifs"
    className="RandomGif-picture"
    src="test-file-stub" />
  <button
    disabled={false}
    onClick={[Function anonymous]}>
    New cat
  </button>
</div>
`;

exports[`test loaded 1`] = `
<div
  className="RandomGif">
  <img
    alt="random gif"
    className="RandomGif-picture"
    src="https://media2.giphy.com/media/m7ychnf9zOVm8/giphy.gif" />
  <button
    disabled={false}
    onClick={[Function anonymous]}>
    New cat
  </button>
</div>
`;

exports[`test loading 1`] = `
<div
  className="RandomGif">
  <img
    alt="waiting logo"
    className="RandomGif-picture RandomGif-picture-loading"
    src="test-file-stub" />
  <button
    disabled={true}
    onClick={[Function anonymous]}>
    Loading...
  </button>
</div>
`;
